apps/s_socket.c

```diff
@@ -283,6 +284,32 @@ int do_server(int *accept_sock, const char *host, const char *port,
     BIO_ADDRINFO_free(res);
     res = NULL;
 
+    if (portfile != NULL) {
+        BIO *pf = NULL;
+        union BIO_sock_info_u info;
+        char *hostname = NULL;
+        char *service = NULL;
+        int success = 0;
+
+        if ((info.addr = BIO_ADDR_new()) != NULL
+            && BIO_sock_info(asock, BIO_SOCK_INFO_ADDRESS, &info)
+            && (hostname = BIO_ADDR_hostname_string(info.addr, 1)) != NULL
+            && (service = BIO_ADDR_service_string(info.addr, 1)) != NULL
+            && (pf = BIO_new_file(portfile, "w")) != NULL
+            && ((strchr(hostname, ':') == NULL
+                 ? /* IPv4 */ BIO_printf(pf, "%s:%s\n", hostname, service)
+                 : /* IPv6 */ BIO_printf(pf, "[%s]:%s\n", hostname, service))
```

Comment:
I think in theory writing a file and reading at the same time creates a race condition.
You could write a file and then rename it to the correct file name, that should be atomic.

Reply:
Considering how it gets used, it gets humorous; create a temp file that then gets renamed... to another temp file.
Sure, it can be done. And that gives us just the more reason to create that temp file filter that's been discussed in #904